File: app/src/test/java/ch/zuehlke/sbb/reddit/data/source/remote/model/posts/RedditPostRedditElementTypeAdapterFactoryTest.java
package ch.zuehlke.sbb.reddit.data.source.remote.model.posts;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.reflect.TypeToken;

import junit.framework.Assert;

import org.junit.Test;

import java.io.IOException;
import java.lang.reflect.Type;
import java.util.List;

import ch.zuehlke.sbb.reddit.AbstractTestCase;

import static ch.zuehlke.sbb.reddit.data.source.remote.RedditElementTypeAdapterFactory.getElementTypeAdapterFactory;

/**
 * Created by chsc on 13.11.17.
 */
public class RedditPostRedditElementTypeAdapterFactoryTest extends AbstractTestCase{


    private static final Type type = new TypeToken<List<RedditPostElement>>() {
    }.getType();


    private static final Gson gson = new GsonBuilder()
            .registerTypeAdapterFactory(getElementTypeAdapterFactory())
            .create();

    @Test
    public void parseComments() throws IOException {
        String jsonFile = readJsonFile("reddit_comments.json");
        Assert.assertNotNull(jsonFile);

        final List<RedditPostElement> redditPostElements = gson.fromJson(jsonFile, type);
        Assert.assertEquals(2,redditPostElements.size());
        Assert.assertTrue(redditPostElements.get(1) instanceof RedditPostElement.DataRedditPostElement);
        RedditPostElement.DataRedditPostElement element = (RedditPostElement.DataRedditPostElement) redditPostElements.get(1);
        Assert.assertEquals(33,element.data.children.size());
        dump(redditPostElements);

    }


}

File: app/src/test/java/ch/zuehlke/sbb/reddit/data/source/remote/model/posts/RedditPostRedditElementTypeAdapterFactoryTest.kt
package ch.zuehlke.sbb.reddit.data.source.remote.model.posts

import ch.zuehlke.sbb.reddit.AbstractTestCase
import ch.zuehlke.sbb.reddit.data.source.remote.RedditElementTypeAdapterFactory.Companion.elementTypeAdapterFactory
import com.google.gson.GsonBuilder
import com.google.gson.reflect.TypeToken
import junit.framework.Assert
import org.junit.Test
import java.io.IOException


/**
 * Created by chsc on 13.11.17.
 */
class RedditPostRedditElementTypeAdapterFactoryTest : AbstractTestCase() {

    @Test
    @Throws(IOException::class)
    fun parseComments() {
        val jsonFile = readJsonFile("reddit_comments.json")
        Assert.assertNotNull(jsonFile)

        val redditPostElements = gson.fromJson<List<RedditPostElement>>(jsonFile, type)
        Assert.assertEquals(2, redditPostElements.size)
        Assert.assertTrue(redditPostElements[1] is RedditPostElement.DataRedditPostElement)
        val element = redditPostElements[1] as RedditPostElement.DataRedditPostElement
        Assert.assertEquals(33, element.data!!.children!!.size)
        dump(redditPostElements)

    }

    companion object {


        private val type = object : TypeToken<List<RedditPostElement>>() {

        }.type


        private val gson = GsonBuilder()
                .registerTypeAdapterFactory(elementTypeAdapterFactory)
                .create()
    }


}